draft-enghardt-panrg-path-properties.md

```diff
@@ -96,42 +96,41 @@ in Path-Aware Networking {{I-D.irtf-panrg-questions}}, which is a product of the
 # Terminology
 
 Node:
-: An entity which processes packets, e.g., sends, receives, forwards, or modifies them.
+: An entity which processes packets, e.g., sends, receives, forwards, or modifies them. A node may be physical or virtual, e.g., a machine or a service function. A node may also be the collection of multiple entities which, as a collection, processes packets, e.g., an entire Autonomous System (AS).
 
 Host:
-: A node that processes packets that are explicitly addressed to itself.
-
-Router:
-: A node that processes packets that are not explicitly addressed to itself.
+: A node that generally executes application programs on behalf of user(s), employing network and/or Internet communication services in support of this function, as defined in {{?RFC1122}}.
 
 Link:
-: A medium or communication facility that connects two or more nodes with each other and enables them to exchange packets. A link can be physical, e.g., a WiFi network which connects an Access Point to stations, or virtual, e.g., a virtual switch which connects two virtual machines hosted on the same physical machine.
+: A medium or communication facility that connects two or more nodes with each other. A link enables a node to send packets to other nodes.
+Links can be physical, e.g., a WiFi network which connects an Access Point to stations, or virtual, e.g., a virtual switch which connects two virtual machines hosted on the same physical machine. A link is unidirectional and bidirectional communication can be modeled as two links between the same nodes in opposite directions.
 
 Path element:
 : Either a node or a link.
 
 Path:
-: A sequence of adjacent path elements, alternating between nodes and links, starting and ending with a host. A path can be viewed as an abstraction on a specific layer, omitting lower layer path elements. For example, a router implementing IPv6 may be a path element on a path when considering the network layer. If this router does not implement transport layer functionality, it is hidden when a higher layer, such as the transport or application layer, is considered. In the case of multicast or broadcast, a single packet may be sent over multiple paths at once -- one path for each combination of sending and receiving host.
+: A sequence of adjacent path elements over which a packet can be transmitted, starting and ending with a node. Note that paths are time-dependent, i.e., the sequence of path elements over which packets are sent between two nodes may change frequently. Furthermore, an entity may have only partial visibility of the path elements that comprise a path.
```

Comment:
I like the idea to change the path definition from "alternating path elements" to "A sequence of adjacent path elements over which a packet can be transmitted" since we address both comments on non-sensible paths (e.g., only links without nodes) and Med's comments on service function chaining.

Is it okay to remove the multicast/broadcast remark from the path definition? It was only there for clarity or do we change how multicast/broadcast is defined?

Should we add a sentence to indicate that we can look at paths using different abstraction levels (e.g., "Also note that different entities might treat path elements at different levels of abstraction."). Although I think this should be clear from the new node definition...

Reply:
About multicast/broadcast: It was only there for clarity, but I think this should be explicitly stated somewhere. I wasn't sure it had to be in the path definition, so I took it out there and added it to my list of "things to check before submitting", so I was planning to add it somewhere else. However, on second thought, maybe the path definition is the best place to add it after all... I think I'll just re-add it to the definition for now. Maybe we'll find a better place for this kinds of clarifications later on - maybe we'll even add a figure? But that's future work.

About the different abstraction levels, similarly, I think this should be somewhere. I'm not quite sure yet where, but I think this is related to partial visibility. The path definition already says "an entity may have only partial visibility of the path elements that comprise a path", so I'll just add something on the abstraction level there for now.

Reply:
I think the updated path definition in 15964a38fd16cadc06a88f04add65487e0499420 looks good now. I'm a bit worried that the abstraction levels are not clearly defined but I also couldn't come up with a better way to describe what we mean...